src/actions/plugins/imswitcher.ts

```diff
@@ -0,0 +1,135 @@
+    }
+    // when you exit from insert-like mode, save origin input method and set it to default
+    if (
+      oldMode === ModeName.Insert ||
+      oldMode === ModeName.SurroundInputMode ||
+      oldMode === ModeName.Replace
+    ) {
+      if (
+        newMode !== ModeName.Insert &&
+        newMode !== ModeName.SurroundInputMode &&
+        newMode !== ModeName.Replace
+      ) {
+        this.switchToDefaultIM();
+        return;
+      }
+    }
+    // when you enter insert-like mode, resume origin input method
+    if (
+      newMode === ModeName.Insert ||
+      newMode === ModeName.SurroundInputMode ||
+      newMode === ModeName.Replace
+    ) {
+      if (
+        oldMode !== ModeName.Insert &&
+        oldMode !== ModeName.SurroundInputMode &&
+        oldMode !== ModeName.Replace
+      ) {
+        this.resumeIM();
+      }
+    }
+  }
+
+  // save origin input method and set input method to default
+  public async switchToDefaultIM() {
+    const obtainIMCmd = configuration.autoSwitchIM.obtainIMCmd;
+    const rawObtainIMCmd = this.getRawCmd(obtainIMCmd);
+    if (obtainIMCmd !== '') {
+      if (existsSync(rawObtainIMCmd)) {
+        const insertIMKey = await this.execShell(obtainIMCmd);
+        if (insertIMKey !== undefined) {
+          this.savedIMKey = insertIMKey.trim();
+        }
+      } else {
+        this.showCmdNotFoundErrorMessage(rawObtainIMCmd);
+      }
+    }
+
+    const defaultIMKey = configuration.autoSwitchIM.defaultIM;
+    if (defaultIMKey !== this.savedIMKey) {
+      this.switchToIM(defaultIMKey);
+    }
+  }
+
+  // resume origin inputmethod
+  public async resumeIM() {
+    const defaultIMKey = configuration.autoSwitchIM.defaultIM;
+    if (this.savedIMKey !== defaultIMKey) {
+      this.switchToIM(this.savedIMKey);
+    }
+  }
+
+  private async switchToIM(imKey: string) {
+    let switchIMCmd = configuration.autoSwitchIM.switchIMCmd;
+    if (!switchIMCmd.includes('{im}')) {
+      this.showWrongSwitchIMCmdErrorMessage();
+      return;
+    }
+    const rawSwitchIMCmd = this.getRawCmd(switchIMCmd);
+    if (switchIMCmd !== '') {
+      if (existsSync(rawSwitchIMCmd)) {
+        if (imKey !== '' && imKey !== undefined) {
+          switchIMCmd = switchIMCmd.replace('{im}', imKey);
+          await this.execShell(switchIMCmd);
+        }
+      } else {
+        this.showCmdNotFoundErrorMessage(rawSwitchIMCmd);
+      }
+    }
+  }
+
+  private getRawCmd(cmd: string): string {
+    const cmds = cmd.split(' ');
+    let rawCmd = '';
+    if (cmds.length > 0) {
+      rawCmd = cmds[0];
+    }
+    return rawCmd;
+  }
+
+  private showCmdNotFoundErrorMessage(cmd: string) {
+    vscode.window.showErrorMessage(
+      'Unable to find ' +
+        cmd +
+        '. check your "vim.autoSwitchIMCommand" in VSCode setting. \
+      Or you can disable "vim.autoSwitchIM" to dismiss this error message'
+    );
+  }
+
+  private showWrongSwitchIMCmdErrorMessage() {
+    vscode.window.showErrorMessage(
```

Comment:
Can you use https://github.com/VSCodeVim/Vim/blob/36bf0a95f80654f47e0ec16794dd245a831fcc85/src/util.ts#L12?